README.md

```diff
@@ -47,7 +47,7 @@ impl HelloService for HelloServer {
 
 fn main() {
     let server_handle = HelloServer.spawn("0.0.0.0:0").unwrap();
-    let client = hello_service::Client::new(server_handle.local_addr()).unwrap();
+    let client = hello_service::Client::new(server_handle.dialer()).unwrap();
```

Comment:
My thoughts were more that the readme example should be the most common simple case. I think it's more common to have a client/server in different processes, rather than in the same one. So, I was thinking that using an address would be more familiar to some people.

A separate issue is that we don't really have a full suite of examples.. Just the benchmark. That should be addressed separately though.

Returning client seems ok too. The one advantage of having a Dialer is that you can send bogus stuff over a stream, which is useful for verifying that we don't misbehave if someone sends junk to the server.